frontend/public/module/k8s/k8s-models.ts

```diff
@@ -72,3 +80,53 @@ export const modelFor = (ref: K8sResourceKindReference) => {
  * NOTE: This will not work for CRDs defined at runtime, use `connectToModels` instead.
  */
 export const allModels = () => k8sModels;
+
+/**
+ * Use this hook to find the model for resources using only group and plural/kind
+ */
+export const useModelFinder = () => {
+  const models: ImmutableMap<string, K8sKind[]> = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'models']),
+  );
+  const pluralsToModelMap = _.values(models.toJS()).reduce((acc, curr) => {
+    const ref = referenceForGroupVersionKind(curr.apiGroup)(curr.apiVersion)(curr.plural);
+    acc[ref] = curr;
+    return acc;
+  }, {});
+  const groupVersionMap: DiscoveryResources['groupVersionMap'] = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'groupToVersionMap']),
+  );
+
+  const findModel = (group: string, kind: string) => {
+    if (!group) {
+      const ref = referenceForGroupVersionKind(group)('v1')(kind);
+      const model = pluralsToModelMap[ref] || models.get(ref);
+      if (model) {
+        return model;
+      }
+    }
+    const preferredVersion = groupVersionMap.getIn([group, 'preferredVersion']);
+    const versions = groupVersionMap.getIn([group, 'versions'])?.toJS();
+    if (preferredVersion) {
+      // Find a model for the CRD that uses this preferred version
+      const ref = referenceForGroupVersionKind(group)(preferredVersion)(kind);
```

Comment:
We shouldn't call `referenceForGroupVersionKind` with something that's not a kind.

Reply:
Creating a new helper called `referenceForGroupVersionPlural`